spec/factories/site.rb

```diff
@@ -1,10 +1,11 @@
 FactoryBot.define do
-  factory :site do
+  factory :site, class: 'TempSite' do
     provider
 
-    initialize_with { Site.find_or_initialize_by(provider: provider, code: code) }
+    initialize_with { TempSite.find_or_initialize_by(provider: provider, code: code) }
```

Comment:
Just hit this while testing, we should amend this to look at uuid instead of code